src/Backend/EdgeTPU/EdgeTPUToolchain.ts

```diff
@@ -0,0 +1,295 @@
+import * as vscode from "vscode";
+import * as ini from "ini";
+import * as fs from "fs";
+import * as path from "path";
+
+import { pipedSpawnSync } from "../../Utils/PipedSpawnSync";
+import { Backend } from "../Backend";
+import { Command } from "../Command";
+import { Compiler } from "../Compiler";
+import { Executor } from "../Executor";
+import { PackageInfo, ToolchainInfo, Toolchains } from "../Toolchain";
+import { DebianToolchain } from "../ToolchainImpl/DebianToolchain";
+import { Version } from "../Version";
+
+class EdgeTPUDebianToolchain extends DebianToolchain {
+  run(cfg: string): Command {
+    let cmd = new Command("edgetpu_compiler");
+    var config = ini.parse(fs.readFileSync(cfg, "utf-8").trim());
+
+    if (config["one-import-edgetpu"] === undefined) {
+      return cmd;
+    }
+
+    let outDir = path.dirname(config["one-import-edgetpu"]["output_path"]);
+    cmd.push("--out_dir");
+    cmd.push(outDir);
+
+    let intermediateTensors =
+      config["one-import-edgetpu"]["intermediate_tensors"];
+    if (intermediateTensors !== undefined) {
+      cmd.push("--intermediate_tensors");
+      cmd.push(intermediateTensors);
+    }
+
+    let showOperations = config["one-import-edgetpu"]["show_operations"];
+    if (showOperations === "True") {
+      cmd.push("--show_operations");
+    }
+
+    let minRuntimeVersion = config["one-import-edgetpu"]["min_runtime_version"];
+    if (minRuntimeVersion !== undefined) {
+      cmd.push("--min_runtime_version");
+      cmd.push(minRuntimeVersion);
+    }
+
+    let searchDelegate = config["one-import-edgetpu"]["search_delegate"];
+    if (searchDelegate === "True") {
+      cmd.push("--search_delegate");
+    }
+
+    let delegateSearchStep =
+      config["one-import-edgetpu"]["delegate_search_step"];
+    if (delegateSearchStep !== undefined) {
+      cmd.push("--delegate_search_step");
+      cmd.push(delegateSearchStep);
+    }
+
+    let inputPath = config["one-import-edgetpu"]["input_path"];
+    cmd.push(inputPath);
+
+    return cmd;
+  }
+}
+
+class EdgeTPUCompiler implements Compiler {
+  private readonly toolchainTypes: string[];
+  private readonly toolchainName: string;
+
+  constructor() {
+    this.toolchainName = "edgetpu-compiler";
+    this.toolchainTypes = ["latest"];
+  }
+
+  getToolchainTypes(): string[] {
+    return this.toolchainTypes;
+  }
+
+  parseVersion(version: string): Version {
+    if (!version.trim()) {
+      throw Error("Invalid version format.");
+    }
+
+    let _version = version;
+    let option = "";
+
+    const optionIndex = version.search(/[~+-]/);
+    if (optionIndex !== -1) {
+      option = version.slice(optionIndex);
+      _version = version.slice(0, optionIndex);
+    }
+
+    const splitedVersion = _version.split(".");
+
+    if (splitedVersion.length > 3) {
+      throw Error("Invalid version format.");
+    }
+
+    let major: number | string;
+    let minor: number | string;
+    let patch: number | string;
```

Comment:
If the patch version is not valid in EdgeTPU, you can remove this line.

Reply:
The latest EdgeTPU Compiler version is 16.0 which has major and minor but old versions like 2.1.302470888 have the patch.
So, I consider the case that the version has the patch.

[EdgeTPU Compiler and Runtime versions](https://coral.ai/docs/edgetpu/compiler/#compiler-and-runtime-versions)